jwst/resample/resample_step.py

```diff
@@ -76,19 +76,18 @@ def process(self, input):
             # resample can only handle 2D images, not 3D cubes, etc
             raise RuntimeError("Input {} is not a 2D image.".format(input_models[0]))
 
-        # Get drizzle parameters reference file
-        self.wht_type = self.weight_type
-        for reftype in self.reference_file_types:
-            ref_filename = self.get_reference_file(input_models[0], reftype)
-
-        if ref_filename != 'N/A':
-            self.log.info('Drizpars reference file: {}'.format(ref_filename))
+        #  Get drizzle parameters reference file, if there is one
+        if 'drizpars' in self.reference_file_types:
+            ref_filename = self.get_reference_file(input_models[0], 'drizpars')
+            self.log.info('Using drizpars reference file: {}'.format(ref_filename))
             kwargs = self.get_drizpars(ref_filename, input_models)
-        else:
-            # If there is no drizpars reffile
-            self.log.info("No DRIZPARS reffile")
+        else:  # no drizpars reference file found
+            ref_filename = 'N/A'
+            self.log.info('No drizpars reference file found.')
             kwargs = self._set_spec_defaults()
 
+        self.wht_type = self.weight_type
+        
```

Comment:
Should we even test that 'drizpars' is in `self.reference_file_types`? We just added 'drizpars' to `self.reference_file_types` on line 52.

The bigger issue is that `'No drizpars reference file found.'` message is printed only when 'drizpars' IS NOT IN `self.reference_file_types` (which is practically never) but not when CRDS returns 'N/A' in this line:

```python
ref_filename = self.get_reference_file(input_models[0], 'drizpars')
```

So, I think a separate `if` is required outside of this `if`:

```python
if ref_filename == 'N/A':
    self.log.info('No drizpars reference file found.')
    kwargs = self._set_spec_defaults()
````